src/solver/conv_ck_igemm_fwd_bias_activ.cpp

```diff
@@ -0,0 +1,421 @@
+#include <ck/tensor_operation/gpu/device/device_conv_fwd_bias_activation.hpp>
+#endif
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS)
+
+using DeviceConvFwdBiasReluPtr = ck::tensor_operation::device::DeviceConvFwdBiasActivationPtr<
+    ck::tensor_operation::element_wise::PassThrough,
+    ck::tensor_operation::element_wise::PassThrough,
+    ck::tensor_operation::element_wise::AddRelu>;
+
+// Forward declare CK's function.
+namespace ck {
+namespace tensor_operation {
+namespace device {
+namespace instance {
+
+void add_device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instances(
+    std::vector<DeviceConvFwdBiasReluPtr>&);
+
+} // namespace instance
+} // namespace device
+} // namespace tensor_operation
+} // namespace ck
+
+namespace miopen {
+namespace solver {
+namespace fusion {
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+struct CKArgs
+{
+    CKArgs(const ProblemDescription& problem)
+    {
+        N        = ProblemInterpreter::GetBatchN(problem);
+        K        = ProblemInterpreter::GetOutputChannelK(problem);
+        C        = ProblemInterpreter::GetInputChannelC(problem);
+        input    = {ProblemInterpreter::GetInputHeightHi(problem),
+                 ProblemInterpreter::GetInputWidthWi(problem)};
+        output   = {ProblemInterpreter::GetOutputHeightHo(problem),
+                  ProblemInterpreter::GetOutputWidthWo(problem)};
+        filter   = {ProblemInterpreter::GetFilterHeightY(problem),
+                  ProblemInterpreter::GetFilterWidthX(problem)};
+        strides  = {ProblemInterpreter::GetAdjustedConvolutionStrideH(problem),
+                   ProblemInterpreter::GetAdjustedConvolutionStrideW(problem)};
+        dilation = {ProblemInterpreter::GetAdjustedConvolutionDilationH(problem),
+                    ProblemInterpreter::GetAdjustedConvolutionDilationW(problem)};
+        lPadding = {ProblemInterpreter::GetInputLeftPadH(problem),
+                    ProblemInterpreter::GetInputLeftPadW(problem)};
+        rPadding = {ProblemInterpreter::GetAdjustedInputRightPadH(problem),
+                    ProblemInterpreter::GetAdjustedInputRightPadW(problem)};
+    }
+    int N;
+    int K;
+    int C;
+    std::vector<int> input;
+    std::vector<int> output;
+    std::vector<int> filter;
+    std::vector<int> strides;
+    std::vector<int> dilation;
+    std::vector<int> lPadding;
+    std::vector<int> rPadding;
+};
+
+std::vector<DeviceConvFwdBiasReluPtr> GetInstances()
+{
+
+    std::vector<DeviceConvFwdBiasReluPtr> op_ptrs;
+    ck::tensor_operation::device::instance::
+        add_device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instances(op_ptrs);
+    return op_ptrs;
+}
+
+template <typename DataType>
+void PerformanceConfigConvCKIgemmFwdBiasActiv::Init(const ProblemDescription& problem)
+{
+    const auto& args = CKArgs{problem};
+    const auto conv  = GetInstances();
+    assert(!conv.empty());
+    this->total_size = conv.size();
```

Comment:
[Important] Is it possible that `conv.size()` will change in the future as CK evolves (causing `add_device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instances()` to change its behavior)?

Reply:
I believe so. As CK evolves `add_device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instances()` could add more kernels.